x-pack/plugins/index_management/server/lib/fetch_indices.ts

```diff
@@ -17,20 +17,69 @@ interface Hit {
   'docs.count': any;
   'store.size': any;
   sth: 'true' | 'false';
+  hidden: boolean;
 }
 
-interface Aliases {
-  [key: string]: string[];
+interface IndexInfo {
+  aliases: { [aliasName: string]: unknown };
+  mappings: unknown;
+  settings: {
+    index: {
+      hidden: 'true' | 'false';
+    };
+  };
 }
 
-interface Params {
-  format: string;
-  h: string;
-  index?: string[];
+interface GetIndicesResponse {
+  [indexName: string]: IndexInfo;
 }
 
-function formatHits(hits: Hit[], aliases: Aliases): Index[] {
-  return hits.map((hit: Hit) => {
+async function fetchIndicesCall(
+  callAsCurrentUser: CallAsCurrentUser,
+  indexNames?: string[]
+): Promise<Index[]> {
+  let indexNamesCSV: string;
+  if (indexNames && indexNames.length) {
+    indexNamesCSV = indexNames.join(',');
+  } else {
+    indexNamesCSV = '*';
+  }
+
+  // This call retrieves alias and settings (incl. hidden status) information about indices
+  const indices: GetIndicesResponse = await callAsCurrentUser('transport.request', {
+    method: 'GET',
+    path: `/${indexNamesCSV}`,
+    query: {
+      expand_wildcards: 'hidden,all',
+    },
+  });
+
+  if (!Object.keys(indices).length) {
```

Comment:
If we know that the `GET /<indices>` is slower than the `_cat/indices`, we might better then fetch first the cat indices and exit early from there. WDYT?

Reply:
I spoke to @danhermann about this, he mentioned that both `_cat/indices` and `/<indices>` would suffer a performance hit but that there is a new endpoint that will support this being built https://github.com/elastic/kibana/issues/64858